src/ooxml/java/org/apache/poi/xddf/usermodel/chart/XDDFBarChartData.java

```diff
@@ -187,5 +187,16 @@ protected CTAxDataSource getAxDS() {
         protected CTNumDataSource getNumDS() {
             return series.getVal();
         }
+        
```

Comment:
overrided method to update series id and order id

Reply:
I can see these methods, but your comment does not help to understand what problem they were supposed to solve in the first place. 

I first of all question their `public` visibility. Since I don't think it is good practice to rely on the developer using the library to **know** when to use and what to expect from them...

Reply:
so we created update methods, because in a chart, we need to provide unique value to each series, now while creating combo chart, i.e. line series and bar series in that case we assigning same values to each series due to that it was corrupting the office file. Now either we can create a new constructor with new parameter i.e. index or can provide an update method to re-assign axis id and order id. so i chose update method.

Reply:
I finally understood what you were trying to achieve! I had a similar code in a PR that I left aside because of some more changes required after a comment about API requiring changes.

https://github.com/apache/poi/pull/139/files#diff-17a515ee7210ab25ee2bcedbf5e49a87R116

Reply:
I also tried to get all series list in chart data, but as there is no relationship between different series types, so we are left with only series list of particular chart data.